mathesar/utils/joins.py

```diff
@@ -16,6 +17,7 @@
 def get_processed_joinable_tables(table, limit=None, offset=None, max_depth=2):
     raw_joinable_tables = ma_sel.get_joinable_tables(
         table.schema._sa_engine,
+        get_empty_metadata(),
```

Comment:
Why not use cached metadata?

Reply:
To be honest, my stance with this PR was "get it working first, reuse / cache later" . 

I want to get things working properly as a base line, then surgically reuse and cache metadata one spot at a time. This will contain any introduced bugs, and help us understand where they were introduced.

Reply:
That makes sense, especially considering that this was urgent.

I'm somewhat concernced that cases where we're not reusing metadata, but could do so, will build until someone will finally be forced to sit down and do a bunch of refactoring.